service/src/main/java/bio/terra/workspace/db/model/DbStateful.java

```diff
@@ -0,0 +1,47 @@
+package bio.terra.workspace.db.model;
+
+import bio.terra.workspace.service.resource.model.WsmResourceState;
+import javax.annotation.Nullable;
+import org.springframework.jdbc.core.namedparam.MapSqlParameterSource;
+
+/** Interface for DB objects that implement states */
+public interface DbStateful {
+  /**
+   * Get the state of the DB object
+   *
+   * @return state
+   */
+  WsmResourceState getState();
+
+  /**
+   * Get the flightId of the DB object
+   *
+   * @return nullable flight id
+   */
+  @Nullable
+  String getFlightId();
+
+  /** Get the text string of the type of object for logging */
+  String getObjectTypeString();
```

Comment:
what about reuse the enum?

Reply:
Which enum? I'm not sure what you mean

Reply:
https://github.com/DataBiosphere/terra-workspace-manager/blob/34b6d06e2d5af4cac1c8c9fde69b84e5301320c6/service/src/main/java/bio/terra/workspace/common/logging/model/ActivityLogChangedTarget.java#L4